docker/src/main/java/brooklyn/entity/container/docker/DockerHostImpl.java

```diff
@@ -591,26 +594,31 @@ public void addIpPermissions(Collection<IpPermission> permissions) {
      * @return Extra IP permissions to be configured on this entity's location.
      */
     protected Collection<IpPermission> getIpPermissions() {
-        String localhost = LocalhostExternalIpLoader.getLocalhostIpWithin(Duration.minutes(1)) + "/32";
-        IpPermission dockerPort = IpPermission.builder()
-                .ipProtocol(IpProtocol.TCP)
-                .fromPort(sensors().get(DockerHost.DOCKER_PORT))
-                .toPort(sensors().get(DockerHost.DOCKER_PORT))
-                .cidrBlock(localhost)
-                .build();
-        IpPermission dockerSslPort = IpPermission.builder()
-                .ipProtocol(IpProtocol.TCP)
-                .fromPort(sensors().get(DockerHost.DOCKER_SSL_PORT))
-                .toPort(sensors().get(DockerHost.DOCKER_SSL_PORT))
-                .cidrBlock(localhost)
-                .build();
+        List<IpPermission> permissions = MutableList.of();
+
+        String publicIpCIDR = LocalhostExternalIpLoader.getLocalhostIpWithin(Duration.minutes(1)) + "/32";
+        permissions.addAll(getClockerPermisionsForCIDR(publicIpCIDR));
+
+        Boolean addLocalhost = config().get(ADD_LOCALHOST_PERMISSION);
+        if(Boolean.TRUE.equals(addLocalhost)) {
```

Comment:
Boolean.TRUE.equals is uglier but prevents null pointer exceptions when someone nulls the config for whatever reason.